caddyconfig/httpcaddyfile/options.go

```diff
@@ -223,17 +223,39 @@ func parseOptSingleString(d *caddyfile.Dispenser) (interface{}, error) {
 }
 
 func parseOptAdmin(d *caddyfile.Dispenser) (interface{}, error) {
-	if d.Next() {
-		var listenAddress string
-		if !d.AllArgs(&listenAddress) {
-			return "", d.ArgErr()
+	adminCfg := new(caddy.AdminConfig)
+	for d.Next() {
+		if d.NextArg() {
+			listenAddress := d.Val()
+			if listenAddress == "off" {
+				adminCfg.Disabled = true
+				if d.Next() { // Do not accept any remaining options including block
+					return nil, d.Err("No more option is allowed after turning off admin config")
+				}
+			} else {
+				adminCfg.Listen = listenAddress
+				if d.NextArg() { // At most 1 arg is allowed
+					return nil, d.Err("At most 1 argument is allowed in admin config")
```

Comment:
This check forbids multiple ips to listen:
```
admin 192.168.1.1:2020 127.0.0.1:2020 {
    enforce_origin
    origins 192.168.1.1:2020 127.0.0.1:2020
}
```

Reply:
You can also return `d.ArgErr()` which is designed for this. It's not required but the error message in this case is very similar anyway.
```suggestion
				adminCfg.Listen = listenAddress
				if d.NextArg() { // At most 1 arg is allowed
					return nil, d.ArgErr()
```